### PR TITLE
Add NFT support for sending and receiving non fungible tokens

### PR DESCRIPTION
**Pull request by judith-Near**
Not pictured, but included in implementation: supports video nfts

Screen Shot 2022-06-14 at 12 46 52 PM

Screen Shot 2022-06-14 at 12 47 07 PM

### CODE REVIEW

**Comment by agileurbanite** [comment]
@marcinbodnar @judith-Near what is the status of this?

File: src/store/multisafe/thunks/onTransferNFT.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+const signTxByLedger = async (
+  nonFungibleTokensService,
+  contract,
+  withApprove,
+  recipientId,
+  multisafeId,
+  state,
+  actions,
+  tokenId,
+  contractName
```
Comment (hcho112) [inline src/store/multisafe/thunks/onTransferNFT.js:14]: Looks like there is too many arguments being passed to a function. Can we rewrite this to pass an object that contains list of above arguments? in that way order of arguments aren't required. 

```
const signTxByLedger = async ({
  nonFungibleTokensService,
  contract,
  withApprove,
  ...
```

File: src/store/multisafe/thunks/onTransferNFT.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  const nonFungibleTokensService = new NonFungibleTokens(near.connection);
+  isNearWallet
+    ? await nonFungibleTokensService.addTransferRequest({
+        multisafeContract: contract, 
+        withApprove,
+        receiverId: recipientId,
+        tokenId, 
+        contractName 
+    })
+    : await signTxByLedger(nonFungibleTokensService, contract, withApprove, recipientId, multisafeId, state, actions, tokenId, contractName);
```
Comment (hcho112) [inline src/store/multisafe/thunks/onTransferNFT.js:54]: following change needed from above comment

**Review by hcho112 — APPROVED** [review]
Left a comment on minor syntax improvement, LGTM

**Review by hcho112 — APPROVED** [review]
LGTM!

File: src/services/NonFungibleTokens.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+import BN from 'bn.js';
+
+const {
+    utils: {
+        format: { parseNearAmount },
+    },
+} = nearApiJs;
+
+// set this to the same value as we use for creating an account and the remainder is refunded
+const NFT_TRANSFER_GAS = parseNearAmount('0.00000000003');
```
Comment (agileurbanite) [inline src/services/NonFungibleTokens.js:11]: Should we export this value to a config?

File: src/services/NonFungibleTokens.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+const ADD_REQUEST_AND_CONFIRM_GAS = new BN('40000000000000')
```
Comment (agileurbanite) [inline src/services/NonFungibleTokens.js:13]: Same here so in the future we just change in one place?

**Review by agileurbanite — APPROVED** [review]
Minor nits mind owning a separate GH issue for those @judith-Near ?

**Pull request merged**